tools/ops/policylambda.py

```diff
@@ -173,32 +174,76 @@ def main():
     for p in collection:
         if p.provider_name != 'aws':
             continue
-        exec_mode_type = p.data.get('mode', {'type': 'pull'}).get('type')
-        if exec_mode_type == 'pull':
-            continue
-
-        sam_func = render(p)
-        if sam_func:
-            sam['Resources'][resource_name(p.name)] = sam_func
-
-            if exec_mode_type == 'config-rule' or exec_mode_type == 'config-poll-rule':
-                configrule_resource = render_config_rule(p)
-                invoke_resource = render_invoke(resource_name(p.name))
-                sam['Resources'][resource_name(p.name) + "Invoke"] = invoke_resource
-                sam['Resources'][resource_name(p.name) + "ConfigRule"] = configrule_resource
-
-            sam_func['Properties']['CodeUri'] = './%s.zip' % p.name
-        else:
-            print("unable to render sam for policy:%s" % p.name)
-            continue
-
-        archive = mu.PolicyLambda(p).get_archive()
+        policy_lambda = dispatch_render(p, sam)
+        archive = policy_lambda.get_archive()
         with open(os.path.join(options.output_dir, "%s.zip" % p.name), 'wb') as fh:
             fh.write(archive.get_bytes())
 
     with open(os.path.join(options.output_dir, 'deploy.yml'), 'w') as fh:
         fh.write(yaml.safe_dump(sam, default_flow_style=False))
 
 
+def test_config_rule_policy():
```

Comment:
Why are the tests included in the primary file?

Reply:
because this is an example script, its not part of a package or even anything we distribute in a release, ie. its not part of the install path, see previous review comments re context. there really isn't a place that tests can be placed outside of this script that this can be imported except as a sidecar in test file in the ops directory, and placing them inline is a choice to avoid cluttering up that directory.

Reply:
i'm back tracking on this, we have pytest configured to only look at modules with test_*.py pattern to speed up test collection, if we want this running in ci, it really needs to be split out to a separate module.